corehq/apps/hqmedia/models.py

```diff
@@ -960,8 +960,10 @@ def get_media_objects(self, build_profile_id=None):
                     media = media.get(map_item.multimedia_id)
                     retry_successes.append(map_item)
                     yield path, media
-                except ResourceNotFound:
+                except ResourceNotFound as e:
                     retry_failures.append(map_item)
+                    if toggles.CAUTIOUS_MULTIMEDIA.enabled(self.domain):
+                        raise e
```

Comment:
will this cause a 500 or is it caught somewhere? Also raising this will skip the soft assert.

Reply:
It'll raise a 500, the ccz will fail and the user will see a generic error.

It will skip the soft assert, but if this happens I think the full stack trace could be useful.